jmx/jmx.go

```diff
@@ -39,10 +39,16 @@ func getCommand(hostname, port, username, password string) []string {
 		cliCommand = []string{jmxCommand}
 	}
 
-	cliCommand = append(
-		cliCommand, "--hostname", hostname, "--port", port,
-		"--username", username, "--password", password,
-	)
+	if username != "" && password != "" {
```

Comment:
There are some values that we want to include always (hostname & port). Then, you could simplify the code having:
```go
cliCommand = append(cliCommand, "--hostname", hostname, "--port", port)
if username != "" && password != "" {
    cliCommand = append(cliCommand, "--username", username, "--password", password)
}
```